### PR TITLE
Consistent error handling

### LINKED ISSUE #87: Consistent exception handling
The exceptions, thrown in the case of a parsing error, should be fit with the already thrown exception. In the case of an un-parsable value (e.g. 'asf' for the elevation), an `IllegalArgumentException` is thrown. This exception is neither documented, nor a good fit into the existing exception strategy.

Use `java.io.InvalidObjectException` for not-parsable values.

### PR DESCRIPTION
**Pull request by Segelzwerg**
fixes #87 changes IllegalArgumentException to java.io.InvalidObjectException and the usages in the projects

### CODE REVIEW

**Comment by jenetics** [comment]
I don't see your point in replacing `IllegalArgumentException` with `IllegalObjectException`. `IOE` is an `IOException`. The places where now an `IOE` is thrown, like the `Longitude` class, has nothing to do with I/O. An `IAE` is the perfect fit. Maybe you can explain your thinking.

**Comment by jenetics** [comment]
OK, this ticket is from myself, ups. I have to think about my thinking ;-=

**Comment by Segelzwerg** [comment]
> OK, this ticket is from myself, ups. I have to think about my thinking ;-)

If I can help, please let me know ;)

**Comment by jenetics** [comment]
I tried to make my intention clearer: https://github.com/jenetics/jpx/issues/87#issuecomment-542797870

**Comment by Segelzwerg** [comment]
I hope I got right now. If not maybe someone else can take offer :D

File: jpx/src/main/java/io/jenetics/jpx/GPX.java [hunk 1]
```diff
@@ -1137,10 +1138,10 @@ public GPX read(final InputStream input)
 					reader.next();
 					return _reader.read(reader, _mode == Mode.LENIENT);
 				} else {
-					throw new IOException("No 'gpx' element found.");
+					throw new InvalidObjectException("No 'gpx' element found.");
 				}
 			} catch (XMLStreamException e) {
-				throw new IOException(e);
+				throw new InvalidObjectException("Invalid 'gpx' input.");
```
Comment (jenetics) [inline jpx/src/main/java/io/jenetics/jpx/GPX.java]: We have to catch also an `IllegalArgumentException` here:
```java
catch (IllegalArgumentException e) {
    throw (InvalidObjectException)new InvalidObjectException(e.getMessage())
        .initCause(e);
}
```

The `fromString` method could also be optimized.
```java
public GPX fromString(final String xml) {
	final byte[] bytes = xml.getBytes();
	final ByteArrayInputStream in = new ByteArrayInputStream(bytes);
	try {
		return read(in);
	} catch (InvalidObjectException e) {
		if (e.getCause() instanceof IllegalArgumentException) {
			throw (IllegalArgumentException)e.getCause();
		}
		throw new IllegalArgumentException(e);
	} catch (IOException e) {
		throw new IllegalArgumentException(e);
	}
}
```

Comment (jenetics) [reply jpx/src/main/java/io/jenetics/jpx/GPX.java]: Yes, having unit tests would be a great idea. :+1:

**Pull request merged**